### ISSUE TITLE
Umlauts not supported as --extra-args

### ISSUE THREAD

**Issue by sebastiangraf** [ansible/tower-cli#232]
We got some issues with umlauts as extra-args in job launches. Even though it works perfectly with multiple parameters, the cli exists with >0 when it comes to umlauts over there.

The following snippet is not working:
```
+ tower-cli job launch ... -extra-vars=version=0.0.0 project=test iib_run_exec=/opt/iib/scripts/run project_email='\''u219237@sbb.ch'\'' the_user_name='\''äöü ÄÖÜ'\'''
*** DETAILS: Getting the record. **********************************************
GET https://deploy.sbb.ch/api/v1/job_templates/1353/
Params: {}

Error: failed to parse some of the extra variables.
variables: 
version=0.0.0 project=test iib_run_exec=/opt/iib/scripts/run project_email='u219237@sbb.ch' the_user_name='äöü ÄÖÜ'
```

**Comment by sebastiangraf** [comment]
Just an addon, with curl, it works like charm

```
+ curl -H 'Content-Type: application/json' -X POST --data '{"extra_vars":{"version":"0.0.0","project":"test","iib_run_exec":"/opt/iib/scripts/run", "project_email":"u219237@sbb.ch", "the_user_name":"äöü ÄÖÜ"}}' -u fs45074:[*******] http://deploy.sbb.ch/api/v1/job_templates/1353/launch/
% Total % Received % Xferd Average Speed Time Time Time Current
Dload Upload Total Spent Left Speed
0 0 0 0 0 0 0 0 -::- -::- -::- 0
100 156 0 0 100 156 0 153 0:00:01 0:00:01 -::- 153
100 3148 100 2992 100 156 2511 130 0:00:01 0:00:01 -::- 2512

Triggert Job:
https://deploy.sbb.ch/#/jobs/3437
Params:
iib_run_exec: /opt/iib/scripts/run
project: test
project_email: "u219237@sbb.ch"
the_user_name: äöü ÄÖÜ
version: 0.0.0
```

**Comment by AlanCoding** [comment]
Thanks for bringing this to our attention. We certainly want to support unicode and this should be very workable with unit tests.

For the specific syntax, I imagine that you want quotes around the data, just like in the curl command:

```
tower-cli job launch -J 22 --extra-vars '{"extra_vars":{"version":"0.0.0","project":"test","iib_run_exec":"/opt/iib/scripts/run", "project_email":"u219237@sbb.ch", "the_user_name":"äöü ÄÖÜ"}}' -v
```

This launches, but the data it sends is:

```
Data: {u'extra_vars': '{"extra_vars": {"project": "test", "version": "0.0.0", "the_user_name": "\\u00e4\\u00f6\\u00fc \\u00c4\\u00d6\\u00dc", "iib_run_exec": "/opt/iib/scripts/run", "project_email": "u219237@sbb.ch"}}'}
```

In Tower, I can see in extra_vars for that job:

```
  the_user_name: äöü ÄÖÜ
```

Is this working correctly for you?

**Comment by sebastiangraf** [comment]
@AlanCoding Thanks for your replay and please excuse my delay. 
It works. Unfortunately, your content comes from variable within a Jenkins Build generating other problems. But this is none of this issue.
Are you planning to fix this bug?

**Comment by AlanCoding** [comment]
I didn't quite have a clear reproducer. But now I think I do, here:

`tower-cli job launch -J 13 --extra-vars="version=0.0.0 project=test iib_run_exec=/opt/iib/scripts/run project_email='u219237@sbb.ch' the_user_name='asdf asdf'" -v`

^ that command launches and puts the extra vars into a dictionary.

`tower-cli job launch -J 13 --extra-vars="version=0.0.0 project=test iib_run_exec=/opt/iib/scripts/run project_email='u219237@sbb.ch' the_user_name='äöü ÄÖÜ'" -v`

^ this command unexpectedly does not launch. It is unexpected when you consider that the only difference from the previous command is the unicode.

Sure, that's a testable bug in the key=value parser that will be on the roadmap to fix.

**Comment by sebastiangraf** [comment]
Thanks for the feedback and the upcoming fix.

**Comment by AlanCoding** [comment]
Fix incoming. This will be released with 3.1.0, looking for a week or so time frame for that.
